util/dockernode.go

```diff
@@ -39,22 +39,34 @@ func (dn *DockerNode) Start(args []string) error {
 	}
 
 	id := bytes.TrimSpace(out)
+	dn.ID = string(id)
 	idfile := filepath.Join(dn.Dir, "dockerID")
 	err = ioutil.WriteFile(idfile, id, 0664)
+
 	if err != nil {
+		killErr := dn.killContainer()
+		if killErr != nil {
+			return combineErrors(err, killErr)
```

Comment:
I still prefer `errors.Wrap` since its a standard we have adopted in other projects, basically something like:
```go
if err = ioutil.WriteFile(idfile, id, 0664); err != nil {
        if killErr := dn.killContainer(); killErr != nil {
                return errors.Wrapf(err, "error killing daemon: %s", killErr)
        }
        return err
}

if err = waitOnAPI(dn); err != nil {
        if killErr := dn.Kill(); killErr != nil {
                return errors.Wrapf(err, "error killing daemon: %s", killErr)
        }
        return err
}

```

Reply:
Right, when I use it like that and force an error for both `err` and `killErr`, I get something like:

```
error killing daemon: exit status 1: Error response from daemon: Invalid signal: IT
: open /test: is a directory
```

That seems confusing and doesn't give the user (or dev) a good understanding of why the error occurred (maybe it makes more sense to you than me, though). I tried switching around the arguments to `Wrapf` to clean up the output, but then the arguments were in a weird/unnatural order. That's when I figured `Errorf` would be the better approach.

Thoughts on all of that? There may simply be a better way to use `Wrapf` in this context, I just wanted to give you all of the details and get your thoughts.

Reply:
No you are right, that does make things confusing. The `combineErrors` method makes failures clear to the user when things go wrong. We should go with that. 

One other alternative could be to use the `errors.WithMessage` method, but i'm happy with either.